pkg/errors/errors.go

```diff
@@ -0,0 +1,44 @@
+package errors
+
+import "fmt"
+
+const (
+	// Value for destinguishing non-codedError type.
+	// Not used by codedError.
+	NotCodedErrorType = iota - 1
+
+	UnknownError // Default value
+	ExistError
+	NotValidError
+
+	// Number of error codes. Add new one before it.
+	// Keep last.
+	ErrorMax
+)
+
+type codedError struct {
+	code int
+	text string
+}
+
+func (e codedError) Error() string {
+	return e.text
+}
+
+// Return error code value or NotCodedErrorType if variable isn't of type codedError
+func ErrorCode(e error) int {
+	custom, ok := e.(*codedError)
+	if !ok {
+		return NotCodedErrorType
+	}
+	return custom.code
+}
+
+// Create new codedError.
+// If code !(NotCodedErrorType < code < ErrorMax) use UnknownError.
+func Errorf(code int, format string, a ...interface{}) error {
+	if code <= NotCodedErrorType || code >= ErrorMax {
```

Comment:
i don't think this should be enforced.
ideally this coded error object can handle an arbitrary set of error codes so it may be possible to import to different projects.

Reply:
This protects from passed wrong code. As an alternative, code can be defined as new type, but in this way it should be exported. What do You think about this?
I don't see any problem with using it in other project and defining new codes.

Reply:
There are no wrong codes :) its just what the application decides to use.
perhaps with the exclusion of a special value (e.g `iota-1`) to mark a not coded error

if i use it in another project by:
```
import "github.com/Mellanox/ib-kubernetes/pkg/errors"

const (
    MyFirstError=iota
    MySecondError
    ...
    MyHunderdError
    MyLastError
)

func foo() error {
    return errors.Errof(MyHunderError, "Whoops something went wrong")
}
```

it will not work with current implemntation right ?

Reply:
> There are no wrong codes

What stops me to create error like this?
`return errors.Errorf(461254, "Error text")`

> it will not work with current implemntation right ?

Almost. Need to modify enums little bit. Like this:
```
const (
    NotCodedErrorType = iota - 1
    MyFirstError
    MySecondError
    ...
    MyHunderdError
    MyLastError
    ErrorMax
)
```

Reply:
> Almost. Need to modify enums little bit. Like this

modify them where ?? the constants are not being overriden

Reply:
>What stops me to create error like this? ....
not the codedError struct

Reply:
> modify them where ?? the constants are not being overriden
> the coded error implementation would still use the constants as defined in `...ib-kubernetes/pkg/errors`

Ah sorry, I thought old enums will be dropped, not extended.

> not the codedError struct

Didn't get what do You mean.

Reply:
its not `Errorf` responsibility to enforce that.